eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ArchiveWSServer.java

```diff
@@ -19,138 +19,86 @@
 
 package uk.ac.ebi.eva.server.ws;
 
-import com.fasterxml.jackson.databind.node.JsonNodeFactory;
-import com.fasterxml.jackson.databind.node.ObjectNode;
 import io.swagger.annotations.Api;
 
 import uk.ac.ebi.eva.commons.mongodb.entities.projections.VariantStudySummary;
 import uk.ac.ebi.eva.lib.utils.QueryResponse;
 import uk.ac.ebi.eva.lib.utils.QueryResult;
-import uk.ac.ebi.eva.lib.utils.QueryOptions;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 import uk.ac.ebi.eva.commons.mongodb.services.VariantStudySummaryService;
-import uk.ac.ebi.eva.lib.metadata.dgva.ArchiveDgvaDBAdaptor;
 import uk.ac.ebi.eva.lib.metadata.eva.ArchiveEvaproDBAdaptor;
-import uk.ac.ebi.eva.lib.metadata.dgva.StudyDgvaDBAdaptor;
+import uk.ac.ebi.eva.lib.metadata.shared.ArchiveWSServerHelper;
 import uk.ac.ebi.eva.lib.metadata.eva.StudyEvaproDBAdaptor;
 import uk.ac.ebi.eva.lib.eva_utils.DBAdaptorConnector;
 import uk.ac.ebi.eva.lib.eva_utils.MultiMongoDbFactory;
+import uk.ac.ebi.eva.lib.utils.QueryUtils;
 
 import java.io.IOException;
-import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 
 @RestController
 @RequestMapping(value = "/v1/meta", produces = "application/json")
 @Api(tags = {"archive"})
 public class ArchiveWSServer extends EvaWSServer {
 
     @Autowired
-    private ArchiveDgvaDBAdaptor archiveDgvaDbAdaptor;
-    @Autowired
     private ArchiveEvaproDBAdaptor archiveEvaproDbAdaptor;
 
     @Autowired
-    private StudyDgvaDBAdaptor studyDgvaDbAdaptor;
-    @Autowired
     private StudyEvaproDBAdaptor studyEvaproDbAdaptor;
+
+    @Autowired
+    private ArchiveWSServerHelper archiveWSServerHelper;
+
+    @Autowired
+    private QueryUtils queryUtils;
```

Comment:
One of the main reasons they were extracted was because there are now two parent WS classes EvaWSServer in eva-server and DgvaWSServer in dgva-server.